app/lib/language_mapper.js

```diff
@@ -16,96 +16,161 @@
+ * @param {string} languageCode 2-letter ISO 639-1 or 3-letter ISO 639-2/T or 3-letter language code, might have optional sub-tags for Script and Region that won't be used. See getLanguageDetails
+ * @param {string} [localeCode='en'] locale/language code to localize language name into
+ * @returns {(string|undefined)} localized language name or undefined if there is no localization available
+ */
+module.exports.getLanguageName = (languageCode, localeCode = 'en') => {
+  if (mappingExistsCache[languageCode] && mappingExistsCache[languageCode][localeCode]) {
+    return mappingExistsCache[languageCode][localeCode];
   }
 
-  getLangs() {
-    if (this.langs == null) {
-      this.langs = require('iso-639-3');
-    }
+  const details = this.getLanguageDetails(languageCode, localeCode);
+  const languageName = details.localized || localeCode === 'en' ? details.languageName : undefined;
 
-    return this.langs;
+  if (!mappingExistsCache[languageCode]) {
+    mappingExistsCache[languageCode] = {};
   }
-
-  mappingMatchesCode(mapping, languageCode) {
-    return (languageCode == mapping.iso6393 ||
-            languageCode == mapping.iso6392B ||
-            languageCode == mapping.iso6392T ||
-            languageCode == mapping.iso6391);
+  mappingExistsCache[languageCode][localeCode] = languageName;
+
+  return languageName;
+};
+
+/**
+ * @typedef {Object} LanguageDetails
+ * @property {string} languageCode Initial normalized 2-letter ISO 639-1 or 3-letter ISO 639-2/T or 3-letter ISO 639-3 language code (without Script or Region sub-tags)
+ * @property {string} languageName Language name localized or in English
+ * @property {(string|undefined)} languageScript Script name based on 4-letter ISO-15924 script sub-tag (i.e. Cyrillic, Arabic, Latin...)
+ * @property {(string|undefined)} languageRegion Region name based on 2-letter ISO 3166-1 region sub-tag (i.e. US, GB...)
+ * @property {boolean} localized Are languageName, languageScript, languageRegion localized
+ * @property {string} type ISO 639-3 language type (i.e. 'living', 'historical', 'constructed')
+ * @property {string} scope ISO 639-3 language scope (i.e. 'individual', 'macrolanguage') See https://en.wikipedia.org/wiki/ISO_639:a
+ * @property {(string|undefined)} iso6393 ISO 693-3 3-letter language code
+ * @property {(string|undefined)} iso6392B ISO 693-2/B 3-letter language code
+ * @property {(string|undefined)} iso6392T ISO 693-2/T 3-letter language code
+ * @property {(string|undefined)} iso6391 ISO 693-1 2-letter language code
+ */
+/**
+ * Function to get various language details
+ * @param {string} languageCode Language identifier is a combination of sub-tags (Language[-Script][-Region]) for Language and optionally Script, and/or Region, according to BCP 47 and RFC 4647
+ * @param {string} [localeCode='en'] Locale/Language code to localize language information into
+ * @returns {LanguageDetails} details Language details 
```

Comment:
I'm not sure if @typedef will look nice for return annotation under JSDoc-generated documentation, but it is picked up nicely by VSCode